src/server/session.ts

```diff
@@ -298,40 +298,55 @@ namespace ts.server {
 
     function combineProjectOutputWhileOpeningReferencedProjects<T>(
         projects: Projects,
+        defaultProject: Project,
         projectService: ProjectService,
         action: (project: Project) => ReadonlyArray<T>,
         getLocation: (t: T) => sourcemaps.SourceMappableLocation,
         resultsEqual: (a: T, b: T) => boolean,
     ): T[] {
         const outputs: T[] = [];
-        combineProjectOutputWorker(projects, <sourcemaps.SourceMappableLocation><any>undefined, projectService, ({ project }, tryAddToTodo) => {
-            for (const output of action(project)) {
-                if (!contains(outputs, output, resultsEqual) && !tryAddToTodo(project, getLocation(output))) {
-                    outputs.push(output);
+        combineProjectOutputWorker(
+            projects,
+            defaultProject,
+            { fileName: "<arbitrary>", position: 0 }, // Location is ignored by callback

```

Comment:
Why cant you pass null instead ? If location is ignored in ```combineProjectOutputWorker```  Dont think you should have need to create default value just for passing sake.